Saving workflow

Reettaphant opened the pull request:
- Saves the metadata and youtube metadata forms only when the form is closed so that we can validate form fields. 

- Using the form validate function in `VideoDisplay.js` to determine when the buttons that close the forms are disabled is not ideal but checking the form validity to determine this would require writing multiple validate functions instead of having one that checks all video fields. If we have time, we should probably spend some time thinking about how we do form validation and perhaps get rid of redux forms.

- The 'create button is disabled in the create form if the atom is not valid

- Only one form can be edited at a time. If edit metadata or edit youtube forms are open, other atom data, including the poster image, cannot be updated: 
<img width="1064" alt="screen shot 2017-02-24 at 09 45 21" src="https://cloud.githubusercontent.com/assets/3066534/23298622/54422e0a-fa76-11e6-87ed-7833caa5be0b.png">


@mbarton @akash1810 @jennysivapalan

akash1810 commented on public/video-ui/src/components/Video/Display.js:
```diff
@@ -62,15 +62,41 @@ class VideoDisplay extends React.Component {
+      youtubeChannel: this.props.video.channelId
+    }));
+
+    if (formFields.some(field => {
```
if `property` != `'metadataEditable'` or  `'youtubeEditable'` it'll be `undefined` and this line will fail.

Could we instantiate `formFields` as `[]`? It also has the side-effect of becoming a `const` too.

akash1810 commented on public/video-ui/src/components/Video/Display.js:
```diff
@@ -62,15 +62,41 @@ class VideoDisplay extends React.Component {
+    if (this.props && this.props.editState[property]) {
       return (
-        <Icon className="icon__done" icon="done" onClick={this.manageEditingState.bind(this, false, property)}/>
+        <button disabled={this.isButtonDisabled(property)} onClick={this.manageEditingState.bind(this, property)}>
```
`.bind` is a little scary...

akash1810 commented on public/video-ui/src/components/VideoMetaData/VideoMetaData.js:
```diff
@@ -8,41 +8,40 @@ import VideoCategorySelect from '../VideoEdit/formComponents/VideoCategory';
 
     return (
-        <div className="form__group">
+        <form className="form__group">
```
why was this change needed?

Reettaphant replied:
It wasn't in the end! I'll change it back.

Reettaphant commented:
@akash1810 I've now updated the styles of the disabled buttons after a chat with @mr-mr , and I also added validation to the create form to disable the create form submission when the atom is invalid to avoid errors there.

akash1810 commented on public/video-ui/src/components/Video/Display.js:
```diff
@@ -8,69 +8,89 @@ import YoutubeMetaData from '../YoutubeMetaData/YoutubeMetaData';
+
+    const formFields = [];
+    if (property === 'metadataEditable') {
+      formFields.splice(0, 0, 'title', 'category');
```
could use `.concat`

akash1810 replied:
`.concat` was bad advice as it returns a new array. The spread operator is the right way to go `formFields.push(...['title', 'category'])`.
